src/models/binary_graph_operations.py

```diff
@@ -0,0 +1,118 @@
+                identified with each other. Touching corners or edges are not counted.
+
+    Returns:
+        np.ndarray: List of node ids in `other_graph` which identify with `node`.
+    """
+
+    # Mode switch
+    assert mode in ["corner", "edge", "interior"]
+    have_valid_overlap = _BULK_SPATIAL_IDENTIFICATION_FUNCTION[mode]
+
+    # Extract relevant node elements for legibility
+    poly = node["geometry"]
+    label = node["class_label"]
+
+    # Get potential candidates for overlap
+    candidate_ids = other_graph.rtree.query(poly, sort=True)
+    # Filter candidates according to the same class label
+    # fmt: off
+    candidate_ids = candidate_ids[
+        other_graph._class_label(candidate_ids) == label  # pylint: disable=protected-access
+    ]
+    # Filter candidates accroding to correct spatial overlap
+    # fmt: on
+    candidate_ids = candidate_ids[
+        have_valid_overlap(
+            poly,
+            other_graph._geometry(candidate_ids),  # pylint: disable=protected-access
+        )
+    ]
+
+    return candidate_ids
+
+
+### Deprecated but kept for tests and backward compatibility
+def identify_node_old(
+    node: dict, other_graph: "GeoGraph", mode: str = "corner"
+) -> List[int]:
+    """
+    Return list of all node ids in `other_graph` which identify with the given `node`.
+
+    Args:
+        node (dict): The node for which to find nodes in `other_graphs` that can be
+            identified with `node`.
+        other_graph (GeoGraph): The GeoGraph object in which to search for
+            identifications
+        mode (str, optional): Must be one of `corner`, `edge` or `interior`. Defaults
+            to "corner".
+            The different modes correspond to different rules for identification:
+            - corner: Polygons of the same `class_label` which overlap, touch in their
+                edges or corners will be identified with each other. (fastest)
+            - edge: Polygons of the same `class_label` which overlap or touch in their
+                edges will be identified with each other.
+            - interior: Polygons of the same `class_label` which overlap will be
+                identified with each other. Touching corners or edges are not counted.
+
+    Returns:
+        List[int]: List of node ids in `other_graph` which identify with `node`.
+    """
+
+    # Mode switch
+    assert mode in ["corner", "edge", "interior"]
+    have_valid_overlap = _SPATIAL_IDENTIFICATION_FUNCTION[mode]
+
+    # Build list of nodes in `other_graph` which identify with `node`
+    identifies_with = []
+    for candidate_id in other_graph.rtree.intersection(node["geometry"].bounds):
+
+        candidate_node = other_graph.df.iloc[candidate_id]
+        have_same_class_label = node["class_label"] == candidate_node["class_label"]
+
+        if have_same_class_label and have_valid_overlap(
+            node["geometry"], candidate_node["geometry"]
```

Comment:
Ah you mean using `node['geometry']` instead of `poly`? 
Sure, if it's that I can easily change that 😉 If it's not, I'm not sure I fully understand what you mean by 
> what `poly` is or check 
?

Reply:
Yeah, just gave this feedback because you specifically wrote in a comment it's for legibility. No need to change it really, it's fine either way.

Reply:
Changed anyways 😉